BUILD.md

```diff
@@ -0,0 +1,25 @@
+# Building Protocol Buffers
+This repository uses generated code via Google's `protoc-gen-go`
+utility, which generates Go code from proto specification.
+
+The source protobufs are found in [protos/](protos/). The transaction
+generics `batch.proto` and `transaction.proto` are sourced from the
+[hyperledger/transact](https://github.com/hyperledger/transact/tree/master/libtransact/protos) repository, whereas `sabre_payload.proto` is
+sourced from [Cargill/splinter](https://github.com/Cargill/splinter/blob/master/examples/gameroom/gameroom-app/sabre_proto/sabre_payload.proto), or alternatively [hyperledger/transact-sdk-go](https://github.com/hyperledger/transact-sdk-javascript/tree/master/protos).
+
+# Build
+**Requirements**
+1. Protobuf Compiler - the protoc binary must be accesible on the build
+`$PATH`. Download the binary from the protocolbuffers/protobuf
+repository [releases](https://github.com/protocolbuffers/protobuf/releases).
+2. Install Golang's `protoc-gen-go` protocol buffers plugin, which is
+maintained as a dependency in go.mod.
+    ```shell
+    go get google.golang.org/protobuf/cmd/protoc-gen-go
+    go install google.golang.org/protobuf/cmd/protoc-gen-go
+    ```
+
+**Build Command**
+```shell
+./scripts/compile_protos.sh
```

Comment:
Following these instructions, the last step results in the following:
```
$ go get google.golang.org/protobuf/cmd/protoc-gen-go
go: downloading google.golang.org/protobuf v1.21.0
go: found google.golang.org/protobuf/cmd/protoc-gen-go in google.golang.org/protobuf v1.21.0
$ go install google.golang.org/protobuf/cmd/protoc-gen-go
$ ./scripts/compile_protos.sh
usage: grep [-abcDEFGHhIiJLlmnOoqRSsUVvwxZ] [-A num] [-B num] [-C[num]]
        [-e pattern] [-f file] [--binary-files=value] [--color=when]
        [--context[=num]] [--directories=action] [--label] [--line-buffered]
        [--null] [pattern] [file ...]
protoc-gen-go: program not found or is not executable
--go_out: protoc-gen-go: Plugin failed with status code 1.
usage: grep [-abcDEFGHhIiJLlmnOoqRSsUVvwxZ] [-A num] [-B num] [-C[num]]
        [-e pattern] [-f file] [--binary-files=value] [--color=when]
        [--context[=num]] [--directories=action] [--label] [--line-buffered]
        [--null] [pattern] [file ...]
protoc-gen-go: program not found or is not executable
--go_out: protoc-gen-go: Plugin failed with status code 1.
usage: grep [-abcDEFGHhIiJLlmnOoqRSsUVvwxZ] [-A num] [-B num] [-C[num]]
        [-e pattern] [-f file] [--binary-files=value] [--color=when]
        [--context[=num]] [--directories=action] [--label] [--line-buffered]
        [--null] [pattern] [file ...]
protoc-gen-go: program not found or is not executable
--go_out: protoc-gen-go: Plugin failed with status code 1.
```

Reply:
Hi @peterschwarz , 
Please see the updated script and let me know if that documentation is sufficient.

Reply:
Looks like an issue with the grep command on mac not matching the arguments in the script.  We can fix that in another PR